python/tank/bootstrap/manager.py

```diff
@@ -202,30 +208,42 @@ def _set_bundle_cache_fallback_paths(self, paths):
     )
 
 
-    def set_progress_callback(self, callback):
+    def set_progress_callback(self, progress_callback):
         """
-        Specify a method to call whenever progress should be reported back.
+        Sets a function to call whenever progress of the bootstrap should be reported back.
+
+        This function should have the following signature::
 
-        The method needs to have the following signature::
+            progress_callback(progress_value, message, current_index, maximum_index)
 
-            progress_callback(message, current_index, max_index)
+        where:
+        - ``progress_value`` is the current progress value, a float number ranging from 0.0 to 1.0
+                             representing the percentage of work completed.
+        - ``message`` is the progress message string to report.
+        - ``current_index`` is an optional current item number being looped over.
+                            This integer number is relative to ``maximum_index``.
+                            Its value is ``None`` when not provided.
+        - ``maximum_index`` is an optional maximum item number being looped over.
+                            This integer number leads to completion of the current progress step.
+                            Its value is ``None`` when not provided.
 
-        The two index parameters are used to illustrate progress
-        over time and looping. ``max_index`` is the total number of
-        current progress items, ``current_index`` is the currently
-        processed item. This can be used to compute a percentage.
-        Note that ``max_index`` may change at any time and is not guaranteed
-        to be fixed.
+        The following old-style signature is also accepted::
```

Comment:
i think we can skip these docs - the code is so new, i doubt anyone is using it yet.